src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/StorageSessionFactory.cs

```diff
@@ -5,13 +5,20 @@
 
     class StorageSessionFactory : ISynchronizedStorage
     {
-        readonly ContainerHolder containerHolder;
-
-        public StorageSessionFactory(ContainerHolder containerHolder)
+        public StorageSessionFactory(ContainerHolder containerHolder, CurrentSharedTransactionalBatchHolder currentSharedTransactionalBatchHolder)
         {
+            this.currentSharedTransactionalBatchHolder = currentSharedTransactionalBatchHolder;
             this.containerHolder = containerHolder;
         }
 
-        public Task<CompletableSynchronizedStorageSession> OpenSession(ContextBag contextBag) => Task.FromResult<CompletableSynchronizedStorageSession>(new StorageSession(containerHolder, contextBag, true));
+        public Task<CompletableSynchronizedStorageSession> OpenSession(ContextBag contextBag)
+        {
+            var storageSession = new StorageSession(containerHolder, contextBag, true);
+            currentSharedTransactionalBatchHolder?.SetCurrent(storageSession);
```

Comment:
Why would this ever be null?

If I am reading `SynchronizedStorage` right, we always create this and pass it into the factory.

https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/35/files#diff-90cae606f9a2e42dd938a294ac38d378R18

Reply:
It is null in the persistence component tests. We don't need this there